modules/analysis-common/src/test/java/org/opensearch/analysis/common/CompoundAnalysisTests.java

```diff
@@ -111,21 +138,38 @@ public Map<String, AnalysisProvider<TokenFilterFactory>> getTokenFilters() {
         }));
     }
 
-    private Settings getJsonSettings() throws IOException {
+    private Path createAndGetHomeDirectory() throws IOException {
+        return createTempDir();
+    }
+
+    private void copyHyphenationPatternsFile(Path home) throws IOException {
+        InputStream hyphenation_patterns_path = getClass().getResourceAsStream("da_UTF8.xml");
+        Path config = home.resolve("config");
+        Files.createDirectory(config);
+        Files.copy(hyphenation_patterns_path, config.resolve("da_UTF8.xml"));
+    }
+
+    private Settings getJsonSettings(Path home) throws IOException {
         String json = "/org/opensearch/analysis/common/test1.json";
         return Settings.builder()
             .loadFromStream(json, getClass().getResourceAsStream(json), false)
             .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
-            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
+            .put(Environment.PATH_HOME_SETTING.getKey(), home.toString())
             .build();
     }
 
-    private Settings getYamlSettings() throws IOException {
+    private Settings getYamlSettings(Path home) throws IOException {
         String yaml = "/org/opensearch/analysis/common/test1.yml";
         return Settings.builder()
             .loadFromStream(yaml, getClass().getResourceAsStream(yaml), false)
             .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
-            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
+            .put(Environment.PATH_HOME_SETTING.getKey(), home.toString())
             .build();
     }
+
+    private Settings[] getSettingsArr() throws IOException {
+        Path home = createAndGetHomeDirectory();
```

Comment:
Why do we create new temp directory for every test case? I believe nothing is changing in the `settingsArr`? Can we consider reusing that?

Reply:
fixed !

Reply:
fixed ! @jainankitk